pkgs/_analyzer_macros/lib/query_service.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:_macro_host/macro_host.dart';
+// ignore: implementation_imports
+import 'package:analyzer/src/summary2/linked_element_factory.dart' as analyzer;
+// ignore: implementation_imports
+import 'package:dart_model/dart_model.dart';
+import 'package:macro_service/macro_service.dart';
+// ignore: implementation_imports
+
+// Hack to access analyzer internals via the introspector that's available.
+// TODO(davidmorgan): handle lifecycle, are these equivalent, do they need
+// tracking per macro application?
+// TODO(davidmorgan): remove hack by injecting an explicit API into the
+// analyzer, once we know what's needed.
+Object? introspector;
```

Comment:
Analyzer implementations should probably be passed in to `AnalyzerMacroImplementation.start` and then passed here.

Reply:
I think we need to avoid using the analyzer instance that we currently create for testing, because in the end the way this code will be integrated it won't control creation of the analyzer instance.

So we need to get the analyzer via the injection. Just, via a better interface than the one that exists now :)